### DISCUSSION TITLE
Show git info for GIT root directories

### DISCUSSION THREAD

**Discussion by Ziwi01** [nvim-tree/nvim-tree.lua#1485] [Q&A] [answered]
Hello,

Is it possible to show GIT root directories status (like branch, behind/ahead) when showing tree from level above? Similarly like IntelliJ IDEA has:
![idea_gitstatus](https://user-images.githubusercontent.com/25208694/182661752-90d14337-0bd0-49b0-a267-6f62350ff1b6.png)

Or from terminal with [LS with GIT status](https://github.com/gerph/ls-with-git-status) - just with arrows instead of literal `{x modified}` to tighten the output.

**Comment by alex-courtis** [comment, marked as answer]
That's outside the scope of nvim-tree; it only shows statuses.

> branch, behind/ahead

[vim-fugitive](https://github.com/tpope/vim-fugitive)'s `:G` should suit your needs. It shows an incredible amount of detail and has an incredible amount of functionality.

> root directories status

[vim-airline](https://github.com/vim-airline/vim-airline) can show a lot of information on your statusline.

That doesn't quite meet your needs out of the box, however if you use `actions.change_dir.global` and other tweaks, you can have nvim-tree change the vim cwd, which airline or some other plugin can show.

**Reply by Ziwi01** [reply]
That's what I thought, thanks for reassurance ;). I already use fugitive and airline, but was wondering if it can be done without changing the working dir or opening file from the project, just by looking on the tree itself. Then I could map a command to maybe run git pull from there on selected folder. But I guess I need to do it the classic way. Cheers!
